configure-hugepages/configure-hugepages.yaml

```diff
@@ -0,0 +1,116 @@
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+# Deploy this DaemonSet to configure hugepages on nodes with the
+# "cloud.google.com/configure-hugepages=true" label.
+#
+# Change the values of NR_HUGEPAGES and HUGEPAGE_SIZE to suit your needs. See
+# https://www.kernel.org/doc/Documentation/vm/hugetlbpage.txt for details.
+# This daemonset currently only supports 1G and 2M hugepages. The task will
+# also fail if the node it's run on does not have enough free memory for the
+# kernel to allocate the requested number of hugepages with the specified size.
+#
+# WARNING: Changing the hugepages requires a kubelet restart. Therefore, in
+# order to avoid disrupting your workloads, it is recommended to create a new
+# node pool with the "cloud.google.com/configure-hugepages=true" label in your
+# cluster, deploy the DaemonSet to configure hugepages in that node pool,
+# and then migrate your workloads to the new node pool.
+
+apiVersion: apps/v1
+kind: DaemonSet
+metadata:
+  name: configure-hugepages
+  namespace: kube-system
+spec:
+  selector:
+    matchLabels:
+      name: configure-hugepages
+  updateStrategy:
+    type: RollingUpdate
+  template:
+    metadata:
+      labels:
+        name: configure-hugepages
+    spec:
+      tolerations:
+      - operator: Exists
+      volumes:
+      - name: host
+        hostPath:
+          path: /
+      hostPID: true
+      initContainers:
+      - name: configure-hugepages
+        image: bash
+        env:
+        # Number of 2M hugepages. Update the value as desired.
+        - name: NR_HUGEPAGES_2M
+          value: "100"
+        # Number of 1G hugepages. Update the value as desired.
+        - name: NR_HUGEPAGES_1G
+          value: "4"
+        command:
+        - /usr/local/bin/bash
+        - -c
+        - |
+          set -euo pipefail
+
+          function err() { echo >&2 "$@"; }
+
+          # Writes to the nr_hugepages sysfs file for the given size (in kB),
+          # and then checks that we got that many hugepages afterward.  Returns
+          # error if the result does not match.
+          #
+          # $1: hugepages size in kB, as seen in /sys/kernel/mm/hugepages.
+          # $2: number of hugepages of that size.
+          function set-nr-hugepages() {
+            [[ "$#" -eq 2 ]] || return
+            local size="$1"; shift
+            local number="$1"; shift
+
+            local dest="/host/sys/kernel/mm/hugepages/hugepages-${size}/nr_hugepages"
+
+            echo "Attempting to create ${number} hugepages of size ${size}"
+            echo -n "${number}" >"${dest}"
+            local actual="$(cat "${dest}")"
```

Comment:
Running this through `shellcheck`:

```
$ shellcheck /tmp/hugepage.sh

In /tmp/hugepage.sh line 18:
  local actual="$(cat "${dest}")"
        ^----^ SC2155 (warning): Declare and assign separately to avoid masking return values.

For more information:
  https://www.shellcheck.net/wiki/SC2155 -- Declare and assign separately to ...
```